packages/react-router/lib/dom/ssr/routeModules.ts

```diff
@@ -96,22 +96,24 @@ export interface LinksFunction {
 
 export interface MetaMatch<
   RouteId extends string = string,
-  Loader extends LoaderFunction | unknown = unknown
+  Loader extends LoaderFunction | ClientLoaderFunction | unknown = unknown
```

Comment:
I noticed during testing that I'm not sure we ever handled `meta` types for `clientLoader`.  I think as long as we always union with `undefined` this should be valid since `meta` would have no data on initial render for `clientLoader`-only routes.

Hybrid routes are trickier because `meta` would have data for the server loader on SSR, and on initial hydration and then would have `clientLoader` data subsequently - but I think that's just a userland solution via `MetaFunction<typeof loader | typeof clientLoader>`